### PR TITLE
Testing To Do list: part 2

### PR DESCRIPTION
**Pull request by NasKhalil**
This branch includes the following features implemented:

- Use jest framework for testing.
- Create a test file addTodo.test.js for a file containing the update, completed status, and clear all completed item functions.
- Group tests using the description() method.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello Team,

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._


Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/clearAllCompleted.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
-export default () => {
-  let data = JSON.parse(localStorage.getItem('todoList'));
-  data = data.filter((elem) => elem.completed !== true);
-  for (let i = 0; i < data.length; i += 1) {
-    data[i].index = i + 1;
+import localStorageMock from './__mocks__/localStorageMock.js';
+
+export default (arr) => {
+  // let data = JSON.parse(localStorage.getItem('todoList'));
```
Comment (mikenath223) [inline src/clearAllCompleted.js:4]: - [x] Kindly remove this commented code.

File: src/updateTodo.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
-  const id = Number(e.target.getAttribute('data-index'));
-  const data = JSON.parse(localStorage.getItem('todoList'));
-  const todoObj = data.find((x) => x.index === id);
+import localStorageMock from './__mocks__/localStorageMock.js';
 
-  todoObj.description = e.target.textContent;
-  localStorage.setItem('todoList', JSON.stringify(data));
+export default (id, arr, str) => {
+  // const id = Number(e.target.getAttribute('data-index'));
+  // const data = JSON.parse(localStorage.getItem('todoList'));
```
Comment (mikenath223) [inline src/updateTodo.js]: - [x] Kindly remove this commented code.

**Comment by NasKhalil** [comment]
### Hi @mikenath223 👋 

Thanks for the feedback, we will update the code now.

Cheers 🚀

**Review by EricMbouwe — APPROVED** [review]
### Hi @NasKhalil, 

# Status: Approved ✔️
Excellent job on this project. Keep it up 👏
Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**